client/src/components/Upload.vue

```diff
@@ -254,12 +271,25 @@ function prepareFiles(files) {
                     val => (val || '').length > 0 || 'This field is required'
                   ]"
                   required
-                  label="Name"
-                  hide-details
-                  :disabled="pendingUpload.uploading"
+                  :label="
+                    (pendingUpload.createFolder ? 'Folder' : 'File') + ' Name'
+                  "
+                  :disabled="
+                    pendingUpload.uploading ||
+                      (!pendingUpload.createFolder &&
+                        pendingUpload.files.length > 1)
+                  "
```

Comment:
I'm not a big fan of logic like this in templates.  I'd rather have this in a computed property with a comment that explains why this condition exists.  I'm not able to read and understand this without a few moments of thought.

`// disable file renaming unless exactly 1 file is being uploaded or after upload is started` is I think the intention here?

@AlmightyYakob thoughts?

Reply:
I'm good with that, there was another thing I removed that I felt might of been better in a computed property.  That was changing the Name input field from "Folder Name" to "File Name" to "File Names" if it was more than one file without a folder.  I took it out because it made the template messy.  I was debating if there was a an internal 'limit' to what should be put in a computed property.

Reply:
I tend to put anything with more than 1 operator in a computed property, and I never put multi-line expressions in templates.

`value > 0` -> fine for template
`value ? 'a' : 'b'` -> refactor.

But that's just me...

Reply:
> @AlmightyYakob thoughts?

I agree, as a rule of thumb, if I have to break a condition/prop value into multiple lines, I'll put it in a computed prop. If it's a small ternary operator, sometimes I'll leave it in.

Reply:
I actually think that only the second part of this needs to be a computed prop: 

`!pendingUpload.createFolder && pendingUpload.files.length > 1`

Because the lines below using this expression will be able to use that as well. Maybe something like `changeableName`, up to you.

Reply:
It's not really a computed a method utilizing the current item should be used.  I need to setup the state of each pendingUpload based on it's properties.  Any thoughts/comments on utilizing a method like 'getFilenameInputState(field, pendingUpload)' which will then return the correct value for the different input fields ('label | 'disabled' | 'hint')?